src/pyuvdata/parameter.py

```diff
@@ -906,47 +934,89 @@ class LocationParameter(UVParameter):
     def __init__(
         self,
         name,
+        *,
         required=True,
         value=None,
         spoof_val=None,
         description="",
-        frame="itrs",
-        ellipsoid=None,
-        acceptable_range=None,
         tols=1e-3,
     ):
         super(LocationParameter, self).__init__(
             name,
             required=required,
             value=value,
             spoof_val=spoof_val,
-            form=3,
             description=description,
-            expected_type=float,
-            acceptable_range=acceptable_range,
+            expected_type=tuple(allowed_location_types),
+            strict_type_check=True,
+            acceptable_range=None,
             tols=tols,
         )
-        self.frame = frame
 
-        if frame == "mcmf" and ellipsoid is None:
-            ellipsoid = "SPHERE"
+    @property
+    def frame(self):
+        """Get the frame."""
+        if isinstance(self.value, EarthLocation):
+            return "itrs"
+        elif hasmoon and isinstance(self.value, MoonLocation):
+            return "mcmf"
+
+    @property
+    def ellipsoid(self):
+        """Get the ellipsoid."""
+        if isinstance(self.value, EarthLocation):
+            return None
+        elif hasmoon and isinstance(self.value, MoonLocation):
+            return self.value.ellipsoid
 
-        self.ellipsoid = ellipsoid
+    def xyz(self):
+        """Get the body centric coordinates in meters."""
+        if self.value is None:
+            return None
+        else:
+            if hasmoon and isinstance(self.value, MoonLocation):
+                centric_coords = self.value.selenocentric
+            else:
+                centric_coords = self.value.geocentric
+            return units.Quantity(centric_coords).to("m").value
```

Comment:
Not asking for a change, but depending on the units of the coords themselves, there may be slight performance gain using `to_value` instead. I think it would take lots of thinking, and even then it just saves a copy. We're not really in a regime where that's the kind of performance we're looking to squeeze out though.

![image](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/assets/8646829/1170e32d-960a-4f45-9b75-4a6896bccd33)

Reply:
Interesting, I wasn't aware of this difference.

Reply:
@mkolopanis I just tried making this switch and all the tests pass on pyuvdata, pyradiosky, pyuvsim, hera_cal and hera_qm with this change in place, so I'm inclined to commit it unless you can think of ways it might break things that aren't covered by our tests.